From d81a2bd5b9d3fab5de952794ddf3f7c7930dbe9d Mon Sep 17 00:00:00 2001
Message-Id: <d81a2bd5b9d3fab5de952794ddf3f7c7930dbe9d.1241177871.git.joerg@alea.gnuu.de>
In-Reply-To: <e879f68a77182a30e0ee2c1ffaea9ae9cd12f326.1241177871.git.joerg@alea.gnuu.de>
References: <e879f68a77182a30e0ee2c1ffaea9ae9cd12f326.1241177871.git.joerg@alea.gnuu.de>
From: =?utf-8?q?J=C3=B6rg=20Sommer?= <joerg@alea.gnuu.de>
Date: Fri, 1 May 2009 13:29:56 +0200
Subject: [PATCH 3/4] Use dpkg-query instead of dpkg

The generic interface for querying the dpkg database is dpkg-query. Dpkg
is only a wrapper for backward support.
---
 checkrestart |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/checkrestart b/checkrestart
index baef62f..20f2121 100755
--- a/checkrestart
+++ b/checkrestart
@@ -132,7 +132,7 @@ def main():
 
     packages = {}
     diverted = None
-    dpkgQuery = ["dpkg", "--search"] + programs.keys()
+    dpkgQuery = ["dpkg-query", "--search"] + programs.keys()
     dpkgProc = subprocess.Popen(dpkgQuery, stdout=subprocess.PIPE, stderr=None,
                                 env = lc_all_c_env)
     for line in dpkgProc.stdout.readlines():
@@ -168,7 +168,7 @@ def main():
     for package in packages.values():
         if package == 'util-linux':
             continue
-        dpkgQuery = ["dpkg", "--listfiles", package.name]
+        dpkgQuery = ["dpkg-query", "--listfiles", package.name]
         dpkgProc = subprocess.Popen(dpkgQuery, stdout=subprocess.PIPE, stderr=None)
         for line in dpkgProc.stdout.readlines():
             path = line[:-1]
@@ -288,7 +288,7 @@ def isdeletedFile (f):
         return 0
     # Skip, if asked to, files that do not belong to any package
     if onlyPackageFiles:
-        dpkgQuery = ["dpkg", "--search", f]
+        dpkgQuery = ["dpkg-query", "--search", f]
         dpkgProc = subprocess.Popen(dpkgQuery, stdout=subprocess.PIPE, stderr=None)
         for line in dpkgProc.stdout.readlines():
             if line.endswith(f):
-- 
1.6.3.rc1

